diff --git a/config/asgi.py b/config/asgi.py
index 886178c..517d395 100644
--- a/config/asgi.py
+++ b/config/asgi.py
@@ -2,8 +2,8 @@ import os
 
 import django
 
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "config.settings.production")
+
 django.setup()
 
 from .routing import application  # noqa
-
-os.environ.setdefault("DJANGO_SETTINGS_MODULE", "config.settings.production")
diff --git a/config/settings/common.py b/config/settings/common.py
index 4759d7a..58777a9 100644
--- a/config/settings/common.py
+++ b/config/settings/common.py
@@ -22,10 +22,7 @@ ROOT_DIR = environ.Path(__file__) - 3  # (/a/b/myfile.py - 3 = /)
 APPS_DIR = ROOT_DIR.path("funkwhale_api")
 
 env = environ.Env()
-try:
-    env.read_env(ROOT_DIR.file(".env"))
-except FileNotFoundError:
-    pass
+env.read_env(env_file=env.str('FUNKWHALE_ENV_FILE', '.env'))
 
 FUNKWHALE_HOSTNAME = None
 FUNKWHALE_HOSTNAME_SUFFIX = env("FUNKWHALE_HOSTNAME_SUFFIX", default=None)
